src/compiler/checker.ts

```diff
@@ -10630,6 +10632,46 @@ namespace ts {
             return false;
         }
 
+        function elaborateArrowFunction(node: ArrowFunction, source: Type, target: Type, relation: Map<RelationComparisonResult>): boolean {
+            // Don't elaborate blocks
+            if (isBlock(node.body)) {
+                return false;
+            }
+            // Or functions with annotated parameter types
+            if (some(node.parameters, ts.hasType)) {

```

Comment:
Why not though?

Reply:
Because we're only validating the return types and the error we may issue on mismatched parameter types is usually better if available.